src/sidebar.c

```diff
@@ -1344,7 +1348,16 @@ static void on_openfiles_document_action(GtkMenuItem *menuitem, gpointer user_da
 	gint action = GPOINTER_TO_INT(user_data);
 
 	if (gtk_tree_selection_get_selected(selection, &model, &iter))
-		on_openfiles_document_action_apply(model, iter, action);
+	{
+		GList *doc_list = on_openfiles_document_action_collect(model, &iter, NULL);
```

Comment:
maybe using a GPtrArray would make more sense, as we don't need the flexibility of a list, and it would reduce the number of allocations -- and it's not likely the number of documents to work on would be huge I'd think.